kong/db/declarative/marshaller.lua

```diff
@@ -0,0 +1,127 @@
+local _M = {}
+
+
+local cjson = require("cjson.safe")
+local tostring = tostring
+local tonumber = tonumber
+local type = type
+local fmt = string.format
+local sub = string.sub
+local cjson_encode = cjson.encode
+local cjson_decode = cjson.decode
+
+
+local TYPES_LOOKUP = {
+    number  = 1,
+    boolean = 2,
+    string  = 3,
+    table   = 4,
+}
+
+
+local marshallers = {
+    shm_value = function(str_value, value_type)
+        return fmt("%d:%s", value_type, str_value)
+    end,
+
+    shm_nil = function(at, ttl)
+        return "0:"
+    end,
+
+    [1] = function(number) -- number
+        return tostring(number)
+    end,
+
+    [2] = function(bool)   -- boolean
+        return bool and "true" or "false"
+    end,
+
+    [3] = function(str)    -- string
+        return str
+    end,
+
+    [4] = function(t)      -- table
+        local json, err = cjson_encode(t)
+        if not json then
+            return nil, "could not encode table value: " .. err
+        end
+
+        return json
+    end,
+}
+
+
+function _M.marshall(value)
+    if value == nil then
+        return marshallers.shm_nil()
+    end
+
+    local value_type = TYPES_LOOKUP[type(value)]
```

Comment:
why is **nil** handled separately?

Reply:
I am not a 100% sure, but that is what's from mlcache so I just copied..

Maybe it is trying to distinguish between `nil` and "this value doesn't exist", it may not be necessary for LMDB use case. Although I don't think we intentionally store nil inside LMDB either as setting a value to `nil` in LMDB is a delete operation. I will remove the logic and see if it still works.

Reply:
Removed and seems to be fine :)